paimon-core/src/main/java/org/apache/paimon/table/system/SnapshotsTable.java

```diff
@@ -207,9 +210,29 @@ public InnerTableRead withFilter(Predicate predicate) {
 
             LeafPredicate snapshotPred =
                     predicate.visit(LeafPredicateExtractor.INSTANCE).get("snapshot_id");
-            if (snapshotPred != null && snapshotPred.function() instanceof Equal) {
-                specificSnapshot = (Long) snapshotPred.literals().get(0);
+            if (snapshotPred != null) {
+                if (snapshotPred.function() instanceof Equal) {
```

Comment:
This will cause performance degradation of equal, because this requires list files

Reply:
Originally I had preserved the efficiency in `snapshotsWithinRange`, with the new code removing the list call it should still be efficient with the iterating on a single snapshopId